pkg/transport/filtering/object_filtering_map.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filtering
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// cleanObjectSpecificsFunction is a function for cleaning fields from a specific object.
+// The function cleans the specific fields in place (object is modified).
+// If the object was retrieved using a lister, it's the caller responsibility
+// to do a DeepCopy before calling this function.
+type cleanObjectSpecificsFunction func(object *unstructured.Unstructured)
+
+func NewObjectFilteringMap() *ObjectFilteringMap {
+	serviceTypeMeta := v1.TypeMeta{Kind: "Service", APIVersion: "v1"}
```

Comment:
FYI: Instead of a literal "v1" here, we could use the result of calling the `String` method of https://github.com/kubernetes/api/blob/v0.28.2/core/v1/register.go#L29

Reply:
good comment. I changed to use directly `corev1.SchemeGroupVersion.WithKind("Service")` in the map as key.